### ISSUE TITLE
Coder as render.com web service

### ISSUE THREAD

**Issue by matifali** [coder/coder#6841]
Deploy Coder to render.com.

- [x] Use Render's managed PostgresDB https://render.com/docs/databases
- [x] Deploy Coder as Web service https://render.com/docs/web-services
- [x] ~~Set up a template using [render terraform provider](https://registry.terraform.io/providers/jackall3n/render/latest)~~
	> They do not have an official terraform provider [yet](https://feedback.render.com/features/p/create-a-terraform-provider-for-renders-iac) and there are issues with the 3rd party version https://github.com/jackall3n/terraform-provider-render/issues/4 
- [x] ~~Set up a Docker template that uses a remote docker daemon~~
	> I will do this as a separate issue. #6921 
- [x] Set up a Render template repo to deploy to Render with one click
- [x] Move 1-click Render deploy repo to github.com/coder/packages 
- [x] ~~Add a Deploy to Render Button in README.md~~
- [x] Test deployment by using Fly.io starter template 
- [ ] Write a blog post on github.com/coder/blogs

**Comment by matifali** [comment]
It looks like if we manually create a database first and then deploy Coder service all works fine, But deploying both of them together as a render blueprint fails,

Some logs,

## First Deployment,

```console
connect to postgres: migrate up: up: migration failed: deadlock detected, Process 182 waits for ShareLock on virtual transaction 7/11; blocked by process 189.
Mar 30 11:38:31 AM  Process 189 waits for ExclusiveLock on advisory lock [16389,0,943348724,1]; blocked by process 182. in line 0: CREATE INDEX CONCURRENTLY workspace_resources_job_id_idx ON workspace_resources USING btree (job_id);
Mar 30 11:38:31 AM   (details: pq: deadlock detected)
```

## Onwards retries,

```console
connect to postgres: migrate up: up: Dirty database version 74. Fix and force version.
```

**Comment by johnstcn** [comment]
It's possible that Render performs some kind of initial migrations on the database deployment (e.g. creating users, databases). This could be a contributing factor for the deadlocks and incomplete database migration. AFAIK we do not run our migrations in a single database transaction?

**Comment by matifali** [comment]
I have tested the deployment with a Fly.io starter template, and everything works fine.

After further investigation, If I create a database manually and then deploy the blueprint, everything works fine. So I guess Render and Coder are trying to run migrations on the database when it is created as part of the render blueprint.

PS. I am still waiting to hear back from render.com support on this.

**Comment by matifali** [comment]
There is an issue with render.com.
On the first service deployment, render initiates two deployments back to back. As the first deployment is interrupted by the 2nd deployment, it leaves the database in a dirty state.

I have raised this issue with Render support, but their email support cycles are slow.
